api/auth/signer/v4/chunked_reader.go

```diff
@@ -37,7 +40,7 @@ var (
 // The chunkedReader returns io.EOF when the final 0-length chunk is read.
 func NewChunkedReader(r io.ReadCloser, streamSigner *ChunkSigner) io.ReadCloser {
 	return &chunkedReader{
-		r: bufio.NewReader(r),
+		r: bufio.NewReaderSize(r, MaxChunkedPayloadLineLength),
```

Comment:
I don't think it's a fix. I can always create something that is larger than this limit and make the gateway panic which is unacceptable.

Reply:
I'm not sure too, but we would try